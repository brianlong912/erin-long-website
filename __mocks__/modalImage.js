module.exports = {
  node: {
    originalAspect: {
      src:
        "/static/05439b0662452aa560316ecc8a7d415d/14b42/50948213126_748ebfeceb_o.jpg",
      base64:
        "data:image/jpeg;base64,/9j/2wBDABALDA4MChAODQ4SERATGCgaGBYWGDEjJR0oOjM9PDkzODdASFxOQERXRTc4UG1RV19iZ2hnPk1xeXBkeFxlZ2P/2wBDARESEhgVGC8aGi9jQjhCY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2P/wgARCAAeABQDASIAAhEBAxEB/8QAGAAAAwEBAAAAAAAAAAAAAAAAAAIDAQT/xAAWAQEBAQAAAAAAAAAAAAAAAAAAAQL/2gAMAwEAAhADEAAAAW49aOggENWc0xAP/8QAHRAAAgICAwEAAAAAAAAAAAAAAQIAAxEhBBITIv/aAAgBAQABBQJNM33YgHXJ9OQwlbVlBh7LVg1KNNZYXOTP/8QAFxEAAwEAAAAAAAAAAAAAAAAAABAREv/aAAgBAwEBPwGGVT//xAAWEQADAAAAAAAAAAAAAAAAAAAAEBH/2gAIAQIBAT8BK//EACAQAAEDAwUBAAAAAAAAAAAAAAABETECIVEDImJxgZH/2gAIAQEABj8C1FXJUsIIPXRKw5yN1l6J+ju5a/gpgk//xAAdEAEBAQEAAgMBAAAAAAAAAAABEQAxIUFRYXGB/9oACAEBAAE/IUPYAt/MwuxO9d6dzeZZw1DOJqYhQGcuD5wwDJ8XWIUfVZ0Sp6dZwBwMBxH93//aAAwDAQACAAMAAAAQ9yhw/8QAGBEAAwEBAAAAAAAAAAAAAAAAAAEREDH/2gAIAQMBAT8QqEDb68f/xAAXEQEBAQEAAAAAAAAAAAAAAAABEQAQ/9oACAECAQE/EKXkDTf/xAAbEAEBAQEBAQEBAAAAAAAAAAABEQAhMVFh4f/aAAgBAQABPxA7l08HOURVIgTk+Di9weA3T4BgAZKj67zcqK8CHd2PIQbf3ULiL5w1TlakaTmu0aqoH5jBALCiTphSBTxH9wcK+CN//9k=",
      aspectRatio: 0.6666666666666666,
      srcSet:
        "/static/05439b0662452aa560316ecc8a7d415d/f836f/50948213126_748ebfeceb_o.jpg 200w,\n/static/05439b0662452aa560316ecc8a7d415d/2244e/50948213126_748ebfeceb_o.jpg 400w,\n/static/05439b0662452aa560316ecc8a7d415d/14b42/50948213126_748ebfeceb_o.jpg 800w,\n/static/05439b0662452aa560316ecc8a7d415d/47498/50948213126_748ebfeceb_o.jpg 1200w,\n/static/05439b0662452aa560316ecc8a7d415d/0e329/50948213126_748ebfeceb_o.jpg 1600w,\n/static/05439b0662452aa560316ecc8a7d415d/daa73/50948213126_748ebfeceb_o.jpg 3648w",
      sizes: "(max-width: 800px) 100vw, 800px",
    },
  },
}