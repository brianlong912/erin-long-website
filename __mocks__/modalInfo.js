module.exports = {
  node: {
    medium: "Oil on canvas",
    description:
      "Butterflies have the ability to fly free, yet grace us with their beauty and visit often",
    title: "Butterflies",
    size: "20cm x 50cm",
  },
}