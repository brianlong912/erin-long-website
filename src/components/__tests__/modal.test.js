import React from "react"
import renderer from "react-test-renderer"
import { render, fireEvent, cleanup } from "@testing-library/react"

import modalImage from "../../../__mocks__/modalImage"
import modalInfo from "../../../__mocks__/modalInfo"
import Modal from "../modal"

afterEach(cleanup)

describe("Modal render", () => {
  it("renders correctly without modalInfo", () => {
    const tree = renderer
      .create(<Modal modalImage={modalImage} />, { attachTo: document.body })
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it("renders correctly with modalInfo", () => {
    const tree = renderer
      .create(<Modal modalImage={modalImage} modalInfo={modalInfo} />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})

test("modal closes with close button", () => {
  //setting up props for render
  var modalVisible = true
  const mockSetModalVisible = jest.fn(value => {
    modalVisible = value
  })
  const { rerender, getByTestId, container } = render(
    <Modal
      modalImage={modalImage}
      modalVisible={modalVisible}
      setModalVisible={mockSetModalVisible}
    />
  )
  
  //expect modal to initially be visible
  expect(container.querySelector("#modal")).toHaveStyle("visibility: visible")

  //Clicking of button
  fireEvent.click(getByTestId("modal-close-button"))
  rerender(
    <Modal
      modalImage={modalImage}
      modalVisible={modalVisible}
      setModalVisible={mockSetModalVisible}
    />
  )
  expect(container.querySelector("#modal")).toHaveStyle("visibility: hidden")
})

test("modal closes with pressing outside of image and info (both present)", () => {
  var modalVisible = true
  const mockSetModalVisible = jest.fn(value => {
    modalVisible = value
  })
  const { rerender, getByTestId, container } = render(
    <Modal
      modalImage={modalImage}
      modalInfo={modalInfo}
      modalVisible={modalVisible}
      setModalVisible={mockSetModalVisible}
    />
  )
  
  expect(container.querySelector("#modal")).toHaveStyle("visibility: visible")

  fireEvent.click(container.querySelector("#modal"))
  rerender(
    <Modal
      modalImage={modalImage}
      modalInfo={modalInfo}
      modalVisible={modalVisible}
      setModalVisible={mockSetModalVisible}
    />
  )
  expect(container.querySelector("#modal")).toHaveStyle("visibility: hidden")
})

test("modal closes with pressing outside of image (no info present)", () => {
  var modalVisible = true
  const mockSetModalVisible = jest.fn(value => {
    modalVisible = value
  })
  const { rerender, getByTestId, container } = render(
    <Modal
      modalImage={modalImage}
      modalVisible={modalVisible}
      setModalVisible={mockSetModalVisible}
    />
  )
  
  expect(container.querySelector("#modal")).toHaveStyle("visibility: visible")

  fireEvent.click(container.querySelector("#modal"))
  rerender(
    <Modal
      modalImage={modalImage}
      modalInfo={modalInfo}
      modalVisible={modalVisible}
      setModalVisible={mockSetModalVisible}
    />
  )
  expect(container.querySelector("#modal")).toHaveStyle("visibility: hidden")
})

test("modal does not close when pressing on image", () => {
  var modalVisible = true
  const mockSetModalVisible = jest.fn(value => {
    modalVisible = value
  })
  const { rerender, getByTestId, container } = render(
    <Modal
      modalImage={modalImage}
      modalInfo={modalInfo}
      modalVisible={modalVisible}
      setModalVisible={mockSetModalVisible}
    />
  )
  
  expect(container.querySelector("#modal")).toHaveStyle("visibility: visible")

  fireEvent.click(container.querySelector("#modal-pic"))
  rerender(
    <Modal
      modalImage={modalImage}
      modalInfo={modalInfo}
      modalVisible={modalVisible}
      setModalVisible={mockSetModalVisible}
    />
  )
  expect(container.querySelector("#modal")).toHaveStyle("visibility: visible")
})

test("modal does not close when pressing info (info present)", () => {
  var modalVisible = true
  const mockSetModalVisible = jest.fn(value => {
    modalVisible = value
  })
  const { rerender, getByTestId, container } = render(
    <Modal
      modalImage={modalImage}
      modalInfo={modalInfo}
      modalVisible={modalVisible}
      setModalVisible={mockSetModalVisible}
    />
  )
  
  expect(container.querySelector("#modal")).toHaveStyle("visibility: visible")

  fireEvent.click(container.querySelector("#modal-info"))
  rerender(
    <Modal
      modalImage={modalImage}
      modalInfo={modalInfo}
      modalVisible={modalVisible}
      setModalVisible={mockSetModalVisible}
    />
  )
  expect(container.querySelector("#modal")).toHaveStyle("visibility: visible")
})

test("modal does not close when pressing info (no info)", () => {
  var modalVisible = true
  const mockSetModalVisible = jest.fn(value => {
    modalVisible = value
  })
  const { rerender, getByTestId, container } = render(
    <Modal
      modalImage={modalImage}
      modalVisible={modalVisible}
      setModalVisible={mockSetModalVisible}
    />
  )
  
  expect(container.querySelector("#modal")).toHaveStyle("visibility: visible")

  fireEvent.click(container.querySelector("#modal-info"))
  rerender(
    <Modal
      modalImage={modalImage}
      modalInfo={modalInfo}
      modalVisible={modalVisible}
      setModalVisible={mockSetModalVisible}
    />
  )
  expect(container.querySelector("#modal")).toHaveStyle("visibility: visible")
})
