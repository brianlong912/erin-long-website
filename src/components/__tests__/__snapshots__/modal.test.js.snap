// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal render renders correctly with modalInfo 1`] = `
<div
  id="modal"
  onClick={[Function]}
  style={
    Object {
      "backgroundColor": "rgba(43, 43, 43, 0.705)",
      "bottom": "0",
      "left": "0",
      "opacity": "0",
      "pointerEvents": "none",
      "position": "fixed",
      "right": "0",
      "top": "0",
      "transition": "opacity 0.4s linear",
      "zIndex": "99",
    }
  }
>
  <button
    data-testid="modal-close-button"
    onClick={[Function]}
    style={
      Object {
        "background": "transparent",
        "border": "white",
        "borderRadius": "50%",
        "color": "inherit",
        "cursor": "pointer",
        "position": "absolute",
        "right": "1.25em",
        "top": "1.25em",
      }
    }
  >
    <svg
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
          "height": "2.5rem",
          "width": "2.5rem",
        }
      }
      viewBox="0 0 352 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
      />
    </svg>
  </button>
  <div
    id="modal-content"
    onClick={[Function]}
    style={
      Object {
        "display": "flex",
        "left": "50%",
        "position": "absolute",
        "top": "50%",
        "transform": "translate(-50%,-50%)",
      }
    }
  >
    <div
      id="modal-pic"
      style={
        Object {
          "width": "0px",
        }
      }
    >
      <div
        className=" gatsby-image-wrapper"
        style={
          Object {
            "maxHeight": null,
            "maxWidth": null,
            "overflow": "hidden",
            "position": "relative",
          }
        }
      >
        <div
          aria-hidden={true}
          style={
            Object {
              "paddingBottom": "150%",
              "width": "100%",
            }
          }
        />
        <img
          alt=""
          aria-hidden={true}
          src="data:image/jpeg;base64,/9j/2wBDABALDA4MChAODQ4SERATGCgaGBYWGDEjJR0oOjM9PDkzODdASFxOQERXRTc4UG1RV19iZ2hnPk1xeXBkeFxlZ2P/2wBDARESEhgVGC8aGi9jQjhCY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2P/wgARCAAeABQDASIAAhEBAxEB/8QAGAAAAwEBAAAAAAAAAAAAAAAAAAIDAQT/xAAWAQEBAQAAAAAAAAAAAAAAAAAAAQL/2gAMAwEAAhADEAAAAW49aOggENWc0xAP/8QAHRAAAgICAwEAAAAAAAAAAAAAAQIAAxEhBBITIv/aAAgBAQABBQJNM33YgHXJ9OQwlbVlBh7LVg1KNNZYXOTP/8QAFxEAAwEAAAAAAAAAAAAAAAAAABAREv/aAAgBAwEBPwGGVT//xAAWEQADAAAAAAAAAAAAAAAAAAAAEBH/2gAIAQIBAT8BK//EACAQAAEDAwUBAAAAAAAAAAAAAAABETECIVEDImJxgZH/2gAIAQEABj8C1FXJUsIIPXRKw5yN1l6J+ju5a/gpgk//xAAdEAEBAQEAAgMBAAAAAAAAAAABEQAxIUFRYXGB/9oACAEBAAE/IUPYAt/MwuxO9d6dzeZZw1DOJqYhQGcuD5wwDJ8XWIUfVZ0Sp6dZwBwMBxH93//aAAwDAQACAAMAAAAQ9yhw/8QAGBEAAwEBAAAAAAAAAAAAAAAAAAEREDH/2gAIAQMBAT8QqEDb68f/xAAXEQEBAQEAAAAAAAAAAAAAAAABEQAQ/9oACAECAQE/EKXkDTf/xAAbEAEBAQEBAQEBAAAAAAAAAAABEQAhMVFh4f/aAAgBAQABPxA7l08HOURVIgTk+Di9weA3T4BgAZKj67zcqK8CHd2PIQbf3ULiL5w1TlakaTmu0aqoH5jBALCiTphSBTxH9wcK+CN//9k="
          style={
            Object {
              "height": "100%",
              "left": 0,
              "objectFit": "cover",
              "objectPosition": "center",
              "opacity": 1,
              "position": "absolute",
              "top": 0,
              "transitionDelay": "500ms",
              "width": "100%",
            }
          }
        />
        <picture>
          <source
            sizes="(max-width: 800px) 100vw, 800px"
            srcSet="/static/05439b0662452aa560316ecc8a7d415d/f836f/50948213126_748ebfeceb_o.jpg 200w,
/static/05439b0662452aa560316ecc8a7d415d/2244e/50948213126_748ebfeceb_o.jpg 400w,
/static/05439b0662452aa560316ecc8a7d415d/14b42/50948213126_748ebfeceb_o.jpg 800w,
/static/05439b0662452aa560316ecc8a7d415d/47498/50948213126_748ebfeceb_o.jpg 1200w,
/static/05439b0662452aa560316ecc8a7d415d/0e329/50948213126_748ebfeceb_o.jpg 1600w,
/static/05439b0662452aa560316ecc8a7d415d/daa73/50948213126_748ebfeceb_o.jpg 3648w"
          />
          <img
            alt=""
            loading="lazy"
            onLoad={[Function]}
            sizes="(max-width: 800px) 100vw, 800px"
            src="/static/05439b0662452aa560316ecc8a7d415d/14b42/50948213126_748ebfeceb_o.jpg"
            srcSet="/static/05439b0662452aa560316ecc8a7d415d/f836f/50948213126_748ebfeceb_o.jpg 200w,
/static/05439b0662452aa560316ecc8a7d415d/2244e/50948213126_748ebfeceb_o.jpg 400w,
/static/05439b0662452aa560316ecc8a7d415d/14b42/50948213126_748ebfeceb_o.jpg 800w,
/static/05439b0662452aa560316ecc8a7d415d/47498/50948213126_748ebfeceb_o.jpg 1200w,
/static/05439b0662452aa560316ecc8a7d415d/0e329/50948213126_748ebfeceb_o.jpg 1600w,
/static/05439b0662452aa560316ecc8a7d415d/daa73/50948213126_748ebfeceb_o.jpg 3648w"
            style={
              Object {
                "height": "100%",
                "left": 0,
                "objectFit": "cover",
                "objectPosition": "center",
                "opacity": 0,
                "position": "absolute",
                "top": 0,
                "transition": "opacity 500ms",
                "width": "100%",
              }
            }
          />
        </picture>
        <noscript
          dangerouslySetInnerHTML={
            Object {
              "__html": "<picture><source srcset=\\"/static/05439b0662452aa560316ecc8a7d415d/f836f/50948213126_748ebfeceb_o.jpg 200w,
/static/05439b0662452aa560316ecc8a7d415d/2244e/50948213126_748ebfeceb_o.jpg 400w,
/static/05439b0662452aa560316ecc8a7d415d/14b42/50948213126_748ebfeceb_o.jpg 800w,
/static/05439b0662452aa560316ecc8a7d415d/47498/50948213126_748ebfeceb_o.jpg 1200w,
/static/05439b0662452aa560316ecc8a7d415d/0e329/50948213126_748ebfeceb_o.jpg 1600w,
/static/05439b0662452aa560316ecc8a7d415d/daa73/50948213126_748ebfeceb_o.jpg 3648w\\" sizes=\\"(max-width: 800px) 100vw, 800px\\" /><img loading=\\"lazy\\" sizes=\\"(max-width: 800px) 100vw, 800px\\" srcset=\\"/static/05439b0662452aa560316ecc8a7d415d/f836f/50948213126_748ebfeceb_o.jpg 200w,
/static/05439b0662452aa560316ecc8a7d415d/2244e/50948213126_748ebfeceb_o.jpg 400w,
/static/05439b0662452aa560316ecc8a7d415d/14b42/50948213126_748ebfeceb_o.jpg 800w,
/static/05439b0662452aa560316ecc8a7d415d/47498/50948213126_748ebfeceb_o.jpg 1200w,
/static/05439b0662452aa560316ecc8a7d415d/0e329/50948213126_748ebfeceb_o.jpg 1600w,
/static/05439b0662452aa560316ecc8a7d415d/daa73/50948213126_748ebfeceb_o.jpg 3648w\\" src=\\"/static/05439b0662452aa560316ecc8a7d415d/14b42/50948213126_748ebfeceb_o.jpg\\" alt=\\"\\" style=\\"position:absolute;top:0;left:0;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center\\"/></picture>",
            }
          }
        />
      </div>
    </div>
    <div
      id="modal-info"
      style={
        Object {
          "backgroundColor": "#2f2f2f",
          "boxSizing": "border-box",
          "display": "block",
          "padding": "20px",
          "width": "18em",
        }
      }
    >
      <div>
        <div>
          Butterflies
        </div>
        <div>
          Oil on canvas
        </div>
        <div>
          20cm x 50cm
        </div>
        <br />
        <div>
          Butterflies have the ability to fly free, yet grace us with their beauty and visit often
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Modal render renders correctly without modalInfo 1`] = `
<div
  id="modal"
  onClick={[Function]}
  style={
    Object {
      "backgroundColor": "rgba(43, 43, 43, 0.705)",
      "bottom": "0",
      "left": "0",
      "opacity": "0",
      "pointerEvents": "none",
      "position": "fixed",
      "right": "0",
      "top": "0",
      "transition": "opacity 0.4s linear",
      "zIndex": "99",
    }
  }
>
  <button
    data-testid="modal-close-button"
    onClick={[Function]}
    style={
      Object {
        "background": "transparent",
        "border": "white",
        "borderRadius": "50%",
        "color": "inherit",
        "cursor": "pointer",
        "position": "absolute",
        "right": "1.25em",
        "top": "1.25em",
      }
    }
  >
    <svg
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
          "height": "2.5rem",
          "width": "2.5rem",
        }
      }
      viewBox="0 0 352 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
      />
    </svg>
  </button>
  <div
    id="modal-content"
    onClick={[Function]}
    style={
      Object {
        "display": "flex",
        "left": "50%",
        "position": "absolute",
        "top": "50%",
        "transform": "translate(-50%,-50%)",
      }
    }
  >
    <div
      id="modal-pic"
      style={
        Object {
          "width": "0px",
        }
      }
    >
      <div
        className=" gatsby-image-wrapper"
        style={
          Object {
            "maxHeight": null,
            "maxWidth": null,
            "overflow": "hidden",
            "position": "relative",
          }
        }
      >
        <div
          aria-hidden={true}
          style={
            Object {
              "paddingBottom": "150%",
              "width": "100%",
            }
          }
        />
        <img
          alt=""
          aria-hidden={true}
          src="data:image/jpeg;base64,/9j/2wBDABALDA4MChAODQ4SERATGCgaGBYWGDEjJR0oOjM9PDkzODdASFxOQERXRTc4UG1RV19iZ2hnPk1xeXBkeFxlZ2P/2wBDARESEhgVGC8aGi9jQjhCY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2P/wgARCAAeABQDASIAAhEBAxEB/8QAGAAAAwEBAAAAAAAAAAAAAAAAAAIDAQT/xAAWAQEBAQAAAAAAAAAAAAAAAAAAAQL/2gAMAwEAAhADEAAAAW49aOggENWc0xAP/8QAHRAAAgICAwEAAAAAAAAAAAAAAQIAAxEhBBITIv/aAAgBAQABBQJNM33YgHXJ9OQwlbVlBh7LVg1KNNZYXOTP/8QAFxEAAwEAAAAAAAAAAAAAAAAAABAREv/aAAgBAwEBPwGGVT//xAAWEQADAAAAAAAAAAAAAAAAAAAAEBH/2gAIAQIBAT8BK//EACAQAAEDAwUBAAAAAAAAAAAAAAABETECIVEDImJxgZH/2gAIAQEABj8C1FXJUsIIPXRKw5yN1l6J+ju5a/gpgk//xAAdEAEBAQEAAgMBAAAAAAAAAAABEQAxIUFRYXGB/9oACAEBAAE/IUPYAt/MwuxO9d6dzeZZw1DOJqYhQGcuD5wwDJ8XWIUfVZ0Sp6dZwBwMBxH93//aAAwDAQACAAMAAAAQ9yhw/8QAGBEAAwEBAAAAAAAAAAAAAAAAAAEREDH/2gAIAQMBAT8QqEDb68f/xAAXEQEBAQEAAAAAAAAAAAAAAAABEQAQ/9oACAECAQE/EKXkDTf/xAAbEAEBAQEBAQEBAAAAAAAAAAABEQAhMVFh4f/aAAgBAQABPxA7l08HOURVIgTk+Di9weA3T4BgAZKj67zcqK8CHd2PIQbf3ULiL5w1TlakaTmu0aqoH5jBALCiTphSBTxH9wcK+CN//9k="
          style={
            Object {
              "height": "100%",
              "left": 0,
              "objectFit": "cover",
              "objectPosition": "center",
              "opacity": 1,
              "position": "absolute",
              "top": 0,
              "transitionDelay": "500ms",
              "width": "100%",
            }
          }
        />
        <picture>
          <source
            sizes="(max-width: 800px) 100vw, 800px"
            srcSet="/static/05439b0662452aa560316ecc8a7d415d/f836f/50948213126_748ebfeceb_o.jpg 200w,
/static/05439b0662452aa560316ecc8a7d415d/2244e/50948213126_748ebfeceb_o.jpg 400w,
/static/05439b0662452aa560316ecc8a7d415d/14b42/50948213126_748ebfeceb_o.jpg 800w,
/static/05439b0662452aa560316ecc8a7d415d/47498/50948213126_748ebfeceb_o.jpg 1200w,
/static/05439b0662452aa560316ecc8a7d415d/0e329/50948213126_748ebfeceb_o.jpg 1600w,
/static/05439b0662452aa560316ecc8a7d415d/daa73/50948213126_748ebfeceb_o.jpg 3648w"
          />
          <img
            alt=""
            loading="lazy"
            onLoad={[Function]}
            sizes="(max-width: 800px) 100vw, 800px"
            src="/static/05439b0662452aa560316ecc8a7d415d/14b42/50948213126_748ebfeceb_o.jpg"
            srcSet="/static/05439b0662452aa560316ecc8a7d415d/f836f/50948213126_748ebfeceb_o.jpg 200w,
/static/05439b0662452aa560316ecc8a7d415d/2244e/50948213126_748ebfeceb_o.jpg 400w,
/static/05439b0662452aa560316ecc8a7d415d/14b42/50948213126_748ebfeceb_o.jpg 800w,
/static/05439b0662452aa560316ecc8a7d415d/47498/50948213126_748ebfeceb_o.jpg 1200w,
/static/05439b0662452aa560316ecc8a7d415d/0e329/50948213126_748ebfeceb_o.jpg 1600w,
/static/05439b0662452aa560316ecc8a7d415d/daa73/50948213126_748ebfeceb_o.jpg 3648w"
            style={
              Object {
                "height": "100%",
                "left": 0,
                "objectFit": "cover",
                "objectPosition": "center",
                "opacity": 0,
                "position": "absolute",
                "top": 0,
                "transition": "opacity 500ms",
                "width": "100%",
              }
            }
          />
        </picture>
        <noscript
          dangerouslySetInnerHTML={
            Object {
              "__html": "<picture><source srcset=\\"/static/05439b0662452aa560316ecc8a7d415d/f836f/50948213126_748ebfeceb_o.jpg 200w,
/static/05439b0662452aa560316ecc8a7d415d/2244e/50948213126_748ebfeceb_o.jpg 400w,
/static/05439b0662452aa560316ecc8a7d415d/14b42/50948213126_748ebfeceb_o.jpg 800w,
/static/05439b0662452aa560316ecc8a7d415d/47498/50948213126_748ebfeceb_o.jpg 1200w,
/static/05439b0662452aa560316ecc8a7d415d/0e329/50948213126_748ebfeceb_o.jpg 1600w,
/static/05439b0662452aa560316ecc8a7d415d/daa73/50948213126_748ebfeceb_o.jpg 3648w\\" sizes=\\"(max-width: 800px) 100vw, 800px\\" /><img loading=\\"lazy\\" sizes=\\"(max-width: 800px) 100vw, 800px\\" srcset=\\"/static/05439b0662452aa560316ecc8a7d415d/f836f/50948213126_748ebfeceb_o.jpg 200w,
/static/05439b0662452aa560316ecc8a7d415d/2244e/50948213126_748ebfeceb_o.jpg 400w,
/static/05439b0662452aa560316ecc8a7d415d/14b42/50948213126_748ebfeceb_o.jpg 800w,
/static/05439b0662452aa560316ecc8a7d415d/47498/50948213126_748ebfeceb_o.jpg 1200w,
/static/05439b0662452aa560316ecc8a7d415d/0e329/50948213126_748ebfeceb_o.jpg 1600w,
/static/05439b0662452aa560316ecc8a7d415d/daa73/50948213126_748ebfeceb_o.jpg 3648w\\" src=\\"/static/05439b0662452aa560316ecc8a7d415d/14b42/50948213126_748ebfeceb_o.jpg\\" alt=\\"\\" style=\\"position:absolute;top:0;left:0;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center\\"/></picture>",
            }
          }
        />
      </div>
    </div>
    <div
      id="modal-info"
      style={
        Object {
          "backgroundColor": "#2f2f2f",
          "boxSizing": "border-box",
          "display": "none",
          "padding": "20px",
          "width": "18em",
        }
      }
    />
  </div>
</div>
`;
